### ISSUE TITLE
Order Item Status to Enable Downloads is set to "Pending," but no download links are presented in "My Downloads" when logged in (fix provided)

### ISSUE THREAD

**Issue by crankycyclops** [magento/magento2#21753]
### Preconditions (*)

1. Magento 2.2.7
2. PHP 7.1

### Steps to reproduce (*)

1. Login to the admin and set Stores -> Configuration -> Catalog -> Downloadable Product Options -> Order Item Status to Enable Downloads to "Pending"
2. Create a user account on the frontend and login
3. Purchase a downloadable product. The product should be placed in the "pending" state after successful completion of payment.
4. Browse to "My Account" -> "My Downloads" and observe that the products, while in pending status and allowed to be downloaded according to the admin setting, do not display download links.

### Expected result (*)

If Stores -> Configuration -> Catalog -> Downloadable Product Options -> Order Item Status to Enable Downloads is set tp "Pending," the "My Downloads" page should contain a list of download links, not just names of the products.

### Actual result (*)

Only the names of the pending products are displayed, and no download product links are displayed beside them.

I've provided a patch to core below that fixes the issue in case it's useful.

### Proposed Solution

I've submitted a pull request which references this issue.

**Comment by magento-engcom-team** [comment]
Hi @crankycyclops. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@crankycyclops do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @shikhamis11 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98725`, `MAGETWO-98726` were created

**Issue Available**: @shikhamis11, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by crankycyclops** [comment]
I'm taking a look at this now. Because I forgot to branch when I opened the original pull request, I've closed it, will fork a fresh copy of the repo, branch, then open another pull request.

**Comment by crankycyclops** [comment]
I didn't branch properly the first time I attempted to fix this issue, so I've forked a fresh copy of the repository and made a new set of pull requests.

**Comment by magento-engcom-team** [comment]
Hi @crankycyclops. Thank you for your report. 
The issue has been fixed in magento/magento2#22073 by @crankycyclops in 2.3-develop branch
Related commit(s):
 - [6076fdcd20ebc1b6e9e1f10b5c99ea4e28f46365](https://github.com/magento/magento2/commit/6076fdcd20ebc1b6e9e1f10b5c99ea4e28f46365)
 - [d32da36e6cbddc176524ac4b68dc6fb78e1eda90](https://github.com/magento/magento2/commit/d32da36e6cbddc176524ac4b68dc6fb78e1eda90)
 - [2c166978e48e502ca86c267e07acc075a1720ec7](https://github.com/magento/magento2/commit/2c166978e48e502ca86c267e07acc075a1720ec7)
 - [5410e077c534127f3492d53f8ab3981691bed2e3](https://github.com/magento/magento2/commit/5410e077c534127f3492d53f8ab3981691bed2e3)
 - [d189b695f831c099cb050585e76c42d3894ca324](https://github.com/magento/magento2/commit/d189b695f831c099cb050585e76c42d3894ca324)
 - [460c4f3778a7f5d8d3a11142dab13b56bdfb2a78](https://github.com/magento/magento2/commit/460c4f3778a7f5d8d3a11142dab13b56bdfb2a78)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @crankycyclops. Thank you for your report. 
The issue has been fixed in magento/magento2#22072 by @crankycyclops in 2.2-develop branch
Related commit(s):
 - [aaae03f8e964c5b097d17b2c74e578ce33b26f80](https://github.com/magento/magento2/commit/aaae03f8e964c5b097d17b2c74e578ce33b26f80)
 - [e644ea7da8381ecc3db0e0d011eee9ef825858d1](https://github.com/magento/magento2/commit/e644ea7da8381ecc3db0e0d011eee9ef825858d1)
 - [19cc7689797171853de564bfc98f74c4062385fd](https://github.com/magento/magento2/commit/19cc7689797171853de564bfc98f74c4062385fd)
 - [1c5c0c86437ed6cc68aeb6ed50b0d12f490c6a27](https://github.com/magento/magento2/commit/1c5c0c86437ed6cc68aeb6ed50b0d12f490c6a27)

The fix will be available with the upcoming 2.2.9 release.
